walnuts_cpp/include/walnuts/walnuts.hpp

```diff
@@ -147,7 +146,7 @@ void macro_step(const F &logp_grad_fun, const Vec<S> &inv_mass, S step,
       }
     }
     if (logp_max - logp_min <= max_error) {
-      // if (true) return;
+      if (true) return;
```

Comment:
We always assume it is reversible?

Reply:
Argh.  No, that's a mistake left over from debugging.  We need to not only call reversible, but that needs to be updated to do the proper check rather than the heuristic I have in place now (which should be close to correct, but potentially biased).  @torekleppe also clued me in to an alternative way to do the thresholding which I will try.

Reply:
I just pushed a change to the branch that fixes this.  It replaces the heuristic reversibility check I had in place with the proper one.  It should actually be easier to read now because I also introduced a helper function that clarifies the structure of the top-level reversibility check.